src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -307,6 +307,20 @@ private static bool IsFloatPointExpressionOfUnknownPrecision(BoundExpression rew
 
                     break;
 
+                case ConversionKind.MethodGroup:
+
+                    if (_inExpressionLambda || explicitCastInCode || (object)symbolOpt == null || !symbolOpt.IsStatic || symbolOpt.MethodKind != MethodKind.Ordinary)
```

Comment:
I'm not so sure about this earlier, and you're probably right, the behavior between explicit and implicit cast should be consistent.